### ISSUE TITLE
build system detection/support buildapi

### ISSUE THREAD

**Issue by cgwalters** [jlebon/redhat-ci#11]
I don't like the `tests` key...let's do something like this:

```
bash-strict: <
  ./autogen.sh --prefix=blah
  make -j 4
  make install
```

etc.

Now, we should also autodetect projects which implement the [build api](https://github.com/cgwalters/build-api).  And hence support an entry like:

```
build:
  config-opts: --enable-foo --disable-bar CFLAGS='fsanitize=undefined`
```

**Comment by jlebon** [comment]
> I don't like the tests key...let's do something like this:

Is it that you don't like that it wants a list, or that the name `tests` is not clear enough? TBH, I find the list way nicer. I was thinking of eventually making them each independent shell invocations (a-la-Makefile). Semantically separating them allows us down the road to have e.g. amount of time per step, or different ways to present the logs, and other cool things! If it's just the name, I'm open to changing that.

> Now, we should also autodetect projects which implement the build api. And hence support an entry like:

OK, that sounds good! I can work on this, unless you already started.

**Comment by cgwalters** [comment]
Right, separating them does allow us to have a UI more like travis'.  Maybe we should take this opportunity to use the exact same keys as travis, i.e. `install:` and `script`.

**Comment by cgwalters** [comment]
I should have elaborated btw, when I said I "didn't like", I think it was more annoying to copy/paste to run locally.  That isn't a huge burden of course.

**Comment by cgwalters** [comment]
Let's just make this issue about buildapi.

**Comment by jlebon** [comment]
Resolved by #13.
